### DISCUSSION TITLE
Support for online InfluxDB (influxdata.com)

### DISCUSSION THREAD

**Discussion by sp4m4r** [arendst/Tasmota#19526] [Ideas]
Dears,

we are using Tasmota since long and one issue we are frequently struggling with is that there is a full support for local InfluxDB (by compiling the respective version) using a local IP and then setting all credentials and information by the command line (like ifxport, ifxhost, ifxheader, ...) - it works like a charm.

However, when it comes to the online influxdb [...].influxdata.com, things are not working at all. Frequently we read about issues with https in the forum, but as we are working on ESP32 this should be no issue? Furthermore we adjusted the original code (sorry for the scrappy photo) in such a way, that https and no port were used anymore to construct the address .

Nothing works. So we are wondering if it wouldn't be easy for you Pro's to adjust things in such a way that if a ESP32 is used and another option like "ifxcloud 1" is set that contruction of the post request is changed in such a way that online influxdb is usable too. This will simplify so much other workflows like "MQTT-NodeRed-Influx-Grafana".

If this topic was discussed elsewhere - we searched around but found nothing comaparable - please excuse.

![image](https://github.com/arendst/Tasmota/assets/74241471/79f70f9b-5691-429c-81a9-70e12386e49e)

Kind regards!

**Comment by barbudor** [comment]
I thought this was addressed this summer by using HttpLight client
But if the url is still built as 'http' in the driver, some updates of the driver is required.

**Reply by sfromis** [reply]
And I'd see no reason to remove the port number from the URL, just use the default port 80 if that is what's relevant.

**Reply by sp4m4r** [reply]
Thank you for your response. I'm not a professional coder, so reading trought the code is challenging to me - so please excuse if understand the code wrong. What I see is that in the [ifx-code](https://github.com/arendst/Tasmota/blob/04e0d23aba1d7fd60064bf84ee367e56e0c1401b/tasmota/tasmota_xdrv_driver/xdrv_59_influxdb.ino#L120C32-L120C32) the url is alway constructed like 
![image](https://github.com/arendst/Tasmota/assets/74241471/50f609dc-8dad-4c84-bdd6-c009acc28c31)

later in the code api and bucket aso is added. 
![image](https://github.com/arendst/Tasmota/assets/74241471/a1637c56-87e9-4125-b681-e6e965893bd7)

We are using influxdata.com in other python projects and the remote address needs to be like below:
![image](https://github.com/arendst/Tasmota/assets/74241471/06ecee0f-5655-4953-95fa-e4637616409c)

Therfore I understand that
1. http is always used instead of https
2. no port should be used, as the urls should look like [...]data.com/api/v2/[...]

**Reply by sfromis** [reply]
Why do you think that no port should be used? Just use default port 80 (or 443) if you have no different one to use.

**Reply by barbudor** [reply]
As a quick test you can try to add the 's' to 'http' and specify port 443 as expected for https

Let us know is this works
I'll look at a more permanent solution next week

**Reply by sp4m4r** [reply]
Dears,
thank you very much for the quick responses. I'll try all suggestions and share the outcome as soon as possible. 

@sfromis as mentioned, I'm just a Hobbyist and as I never had to declare a port in the python-projects I expected there has to be none at all for the cloud-influx. Thank you for clarifying.

**Reply by barbudor** [reply]
Actually you shouldn't have to specify a port if it is standard (80 for http, 443 for https)
The solution should pickup the right one and you would have to specify only in case of specific variants

I'll look also at making the port optional

**Reply by sfromis** [reply]
Sure, it's correct not *having* to specify a port, but not a requirement to be able to remove it. That was what I was trying to address.

**Reply by sp4m4r** [reply]
As mentioned, I changed the code as below:

```c
IFDB._serverUrl = "https://";
IFDB._serverUrl += ifdb_ip.toString();
IFDB._serverUrl += ":";
IFDB._serverUrl += Settings->influxdb_port;
return true;
```
Then, after compilation and setup and initialization I obtained the output below. (ifx-settings are shown in the first line). It is the first time I receive a respose, so I feel we are moving forward.

![image](https://github.com/arendst/Tasmota/assets/74241471/9809ec13-2e7d-4cc3-a08a-e6b6371fcc24)

**Reply by sp4m4r** [reply]
While checking the console output I found one more interesting point - frequently ifx is checking the server for health, as in the image below.
![image](https://github.com/arendst/Tasmota/assets/74241471/675e807c-c0d7-4ca8-b76b-50890cd0d3a6)

It appears, that the resolved IP "35.202.191.0" is leading to that 404 page. It seems that the IP is obtained from the full url, including https://, so maybe there is a problem by doubling https:// in the final query.

![image](https://github.com/arendst/Tasmota/assets/74241471/226dc245-36de-4f19-8c2b-96262476b1a6)

**Reply by Brightness248** [reply]
Hello all,

Are there any new findings here already? 
I am very interested in this topic. I already faced the same problem.

**Reply by barbudor** [reply]
I'm currently looking at it

**Reply by barbudor** [reply]
PR #19582 on it's way. 
Doc will be updated after merge.

**Reply by Brightness248** [reply]
I am thrilled!
Thank you very much.
I've been a huge fan of tasmota for years. You all do a great job.

Thanks again to all of you!

**Reply by barbudor** [reply]
PR merged, now be available in latest development version of tasmota32
Docs also updated

**Reply by sp4m4r** [reply]
Thank you so much. We've tested the feature (thanks to updated docs it was really simple) and everything works like a charm.
